content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+const myably = new ably('XXX_API_KEY')
+//set the channel you will like to publish to 
+const channel = myably.channels.get('cog-tie');
+//set cors middleware
+app.use(function(req, res, next) {
+res.header("Access-Control-Allow-Origin", "*");
+res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+next();
+});
+//handle route postfunction
+app.post('/', function(req, res) {
+var message = {
+    user: req.body.name,
+    msg: req.body.msg,
+    action: req.body.action
+}
+//Publish your message
+channel.publish('message', message, function(err) {
+    if (err) {
+        console.log('publish failed with error ' + err);
+    } else {
+        console.log('publish succeeded ' + message.msg);
+    }
+})
+res.send({ status: 'okay', message: message })
+
+})
+
+//listen on port and serve the app
+app.listen(3000, function() {
+console.log('Example app listening on port 3000!')
+})
+```
+The code block above is our Express API server setup. 
+
+At the beginning of the file, you had required Express, Body-parser and Ably libraries for Node.js respectively. 
+Next, you initialized a new Pusher object, passing in your Api Key.
+Next, you set the CORS header to your request, using a middleware function.
+Finally, you create a post handler for the @\@ route, and we then make  @Ably@ publish to a channel called @cog-tie@ with an event called @message@.
+Note both the channel name and the event name used in this server. The channel name will be subscribed to while you will listen for the event in your React Native app.
+This is all we need at the server side for your API call to work.
+
+Next, start up your API by running:
+
+```[sh]
+node server.js
+```
+
+h2. Building The React Native Application
+
+To get started, create a new file called @chat.js@ and add paste in the following:
+
+```[javascript]
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ * @flow
+ */
+import React, { Component } from 'react';
+import { StyleSheet, View } from 'react-native'
+export default class Chat extends Component<{}> {
+  render() {
+    return (
+      <View style={styles.container}>
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#F5FCFF',
+    paddingTop: 10,
+  },
+});
+```
+In the code above, nothing much is happening. It is a blank page which you will use to hold your chat component later on.
+
+Next, Open @App.js@ and replace content with:
+
+```[javascript]
+/**
+* Sample React Native App
+* https://github.com/facebook/react-native
+* @flow
+*/
+import React, { Component } from 'react';
+import Chat from './Chat'
+import {
+StyleSheet,
+View,
+Text,
+TextInput,
+Button,
+AsyncStorage,
+} from 'react-native';
+
+export default class App extends Component<{}> {
+constructor(props) {
+super(props);
+this.state={txt:'', user: null}
+this.handler = this.handler.bind(this)
+}
+
+componentDidMount = () =>{
```

Comment:
I don't follow why you're using `ECMA6` here with => and functions in the server code.  Can we be consistent please.

Reply:
Sorry about the mixup. I have changed everything to `ECMA6`